var/spack/repos/builtin/packages/hdf/package.py

```diff
@@ -171,6 +171,10 @@ def configure_args(self):
                 'FCFLAGS=-fallow-argument-mismatch']
             )
 
+        # https://forum.hdfgroup.org/t/help-building-hdf4-with-clang-error-implicit-declaration-of-function-test-mgr-szip-is-invalid-in-c99/7680
+        if self.spec.satisfies('%apple-clang'):
+            config_args.append('CFLAGS=-Wno-error=implicit-function-declaration')
+
```

Comment:
This problem has been fixed in what will be the next release (4.2.16).  When that release is added we can make this conditional for all previous releases.

Reply:
Glad to know that HDF4 is still being maintained and getting new bugfix releases!